src/pages/UserPage/UserPage.jsx

```diff
@@ -1,96 +1,204 @@
+import React, { useState, useEffect } from 'react';
 import Header from '../../components/HeaderComponent/HeaderComponent';
 import Footer from '../../components/Footer/Footer';
 import './UserPage.css';
-import userDB from '../../assets/admins.json';
 import { RiEdit2Fill } from 'react-icons/ri'
-
-
+import { useParams } from 'react-router';
+import languageSelector from '../../assets/languages/languageSelector';
+import { useSelector, useDispatch } from 'react-redux';
+import loadingGif from '../../assets/fotos/loading/loading-gif.gif'
+import axios from 'axios';
+import {
+  fetchUserRequest,
+  fetchUserSuccess,
+  fetchUserFailure,
+  updateUserRequest,
+  updateUserSuccess,
+  updateUserFailure
+} from '../../store/actions/user.action';
 
 const UserPage = () => {
-  const [user, setUser] = useState(userDB[0]);
+  const { id } = useParams();
+  const language = useSelector(state => state.languageReducer);
+  const loading = useSelector((state) => state.userReducer.loading);
+  const user = useSelector(state => state.userReducer.user);
   const [isEditable, setIsEditable] = useState(false);
   const [errors, setErrors] = useState({});
+  const dispatch = useDispatch();
+  
   const [formData, setFormData] = useState({
-    name: user.name,
-    email: user.email,
-    password: user.password,
-    phoneNumber: user.phoneNumber,
-    address: user.address,
-    city: user.city
+    name: '',
+    last_name: '',
+    email: '',
+    password: '',
+    phone_numbers: [],
+    address: [],
+    city: ''
   });
 
+
+  // does a get request on mount 
+  useEffect(() => {
+    const fetchUser = async () => {
+      dispatch(fetchUserRequest());
+      try {
+        const response = await axios.get(`http://localhost:8080/api/users/${id}`);
+        dispatch(fetchUserSuccess(response.data.data));
+        setFormData({
+          name: response.data.data.name,
+          last_name: response.data.data.last_name,
+          email: response.data.data.email,
+          password: response.data.data.password,
+          phone_numbers: response.data.data.phone_numbers,
+          address: response.data.data.addresses,
+          city: response.data.data.city,
+        });
+      } catch (error) {
+        dispatch(fetchUserFailure(error));
+      }
+    };
+    fetchUser();
+  }, [id, dispatch]);
+
+  // enables the editing of the inputs
   const handleEditClick = () => {
-    setIsEditable(!isEditable)
+    setIsEditable(!isEditable);
   }
+
+  // hangles the changes of the inputs
   const handleInputChange = (event) => {
     const { name, value } = event.target;
-    setFormData((prevState) => ({
-      ...prevState,
-      [name]: value,
-    }));
-  }
-  
-  const handleSumbit = (event) => {
+    const matches = name.match(/^(\w+)\[(\d+)\](\[\w+\])?$/);
+    if (matches) {
+      const fieldName = matches[1];
+      const fieldIndex = parseInt(matches[2], 10);
+      const nestedFieldName = matches[3] ? matches[3].slice(1, -1) : undefined;
```

Comment:
Consider create a variable that represents the array positions for matches, would be better for reading and understanding.

Reply:
Resolved

Reply:
I was expecting an object, but indexes variables clarifies the meaning, nice for it!